providers/dns/nifcloud/nifcloud.go

```diff
@@ -0,0 +1,85 @@
+// Package nifcloud implements a DNS provider for solving the DNS-01 challenge
+// using NIFCLOUD DNS.
+package nifcloud
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/xenolf/lego/acme"
+)
+
+// DNSProvider implements the acme.ChallengeProvider interface
+type DNSProvider struct {
+	client *NifcloudDNSClient
+}
+
+// NewDNSProvider returns a DNSProvider instance configured for the NIFCLOUD DNS service.
+// Credentials must be passed in the environment variables: NIFCLOUD_ACCESS_KEY_ID and NIFCLOUD_SECRET_ACCESS_KEY.
+func NewDNSProvider() (*DNSProvider, error) {
+	client := NewDNSClient()
+
+	return &DNSProvider{
+		client: client,
+	}, nil
+}
```

Comment:
Could rewrite like the following example:
https://github.com/xenolf/lego/blob/c4bbb4b819c087d02066fe745d7ca5b60a480d36/providers/dns/cloudflare/cloudflare.go#L32-L39